### PR TITLE
Ensure $request->hasSession() is always checked before calling getSession() 

### PR DESCRIPTION
**Pull request by Arman-Hosseini**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Any reason why you're proposing thing?

**Comment by Arman-Hosseini** [comment]
@nicolas-grekas 
Just improving the codes.

**Comment by fabpot** [comment]
We do not change code that works. We did such refactoring in the past and that introduced some bugs. So, we either refactor to fix a bug or we keep the current code.

**Review by nicolas-grekas — APPROVED** [review]
I rebased this PR for 4.4 and reviewed the full code base to ensure we don't call getSession without checking hasSession before, as it is required now (getSession throws in 5.0 instead of returning null)

File: src/Symfony/Bridge/Twig/AppVariable.php [hunk 1]
```diff
@@ -157,8 +156,7 @@ public function getDebug()
     public function getFlashes($types = null)
     {
         try {
-            $session = $this->getSession();
-            if (null === $session) {
+            if (null === $session = $this->getSession();
```
Comment (xabbuh) [inline src/Symfony/Bridge/Twig/AppVariable.php]: `if (null === $session = $this->getSession()) {`

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/Twig/AppVariable.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Bridge/Twig/AppVariable.php]: fixed thanks

**Comment by Arman-Hosseini** [comment]
@nicolas-grekas thank you.

**Review by chalasr — APPROVED** [review]

**Review by xabbuh — DISMISSED** [review]
looks like some tests are failing

**Comment by chalasr** [comment]
@xabbuh fixed, thanks

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Arman-Hosseini.

**Pull request merged**